smartling-api-commons/src/main/java/com/smartling/resteasy/ext/ExtendedMultipartFormWriter.java

```diff
@@ -0,0 +1,105 @@
+package com.smartling.resteasy.ext;
+
+import org.jboss.resteasy.annotations.providers.multipart.PartType;
+import org.jboss.resteasy.plugins.providers.multipart.FieldEnablerPrivilegedAction;
+import org.jboss.resteasy.plugins.providers.multipart.MultipartFormAnnotationWriter;
+import org.jboss.resteasy.plugins.providers.multipart.MultipartFormDataOutput;
+import org.jboss.resteasy.spi.WriterException;
+
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.ext.Provider;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.security.AccessController;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Provides additional functionality for {@code @MultipartForm} annotated parameter objects.
+ *
+ * <ul>
+ *     <li>Extracts any {@link Map} parameter annotated with {@link DynamicFormParam} and {@link PartType}
+ *     into a number of multipart fields.</li>
+ *     <li>Extracts any {@link List} annotated with {@link ListFormParam} and {@link PartType}
+ *     into an array of multipart form fields.</li>
+ * </ul>
+ *
+ */
+@Provider
+@Produces("multipart/form-data")
+public class ExtendedMultipartFormWriter extends MultipartFormAnnotationWriter
+{
+    @Override
+    protected void getFields(Class<?> type, MultipartFormDataOutput output, Object obj) throws IOException
+    {
+        for (Field field : type.getDeclaredFields())
+        {
+            if (field.isAnnotationPresent(DynamicFormParam.class) && field.isAnnotationPresent(PartType.class))
+            {
+                AccessController.doPrivileged(new FieldEnablerPrivilegedAction(field));
+
+                PartType partType = field.getAnnotation(PartType.class);
```

Comment:
Can it be defined one at the beginning of loop?

Reply:
sure, done